core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala

```diff
@@ -83,18 +83,22 @@ private[spark] object ExecutorsPage {
     val memUsed = status.memUsed
     val maxMem = status.maxMem
     val diskUsed = status.diskUsed
-    val totalCores = listener.executorToTotalCores.getOrElse(execId, 0)
-    val maxTasks = listener.executorToTasksMax.getOrElse(execId, 0)
-    val activeTasks = listener.executorToTasksActive.getOrElse(execId, 0)
-    val failedTasks = listener.executorToTasksFailed.getOrElse(execId, 0)
-    val completedTasks = listener.executorToTasksComplete.getOrElse(execId, 0)
+    val totalCores = listener.executorToTaskSummary.get(execId).map(_.totalCores).getOrElse(0)
```

Comment:
Likewise it's probably easier to get this once, and then possibly handle the case where it doesn't exist once? In that case you return an ExecutorSummary with a lot of simple default values. Might be simpler.